src/sip/reply.c

```diff
@@ -16,6 +16,37 @@
 #include "sip.h"
 
 
+#define DEBUG_MODULE "reply"
+#define DEBUG_LEVEL 5
+#include <re_dbg.h>
+
+
+#if HAVE_INET6
+static int dst_set_scopeid(struct sa *dst, struct sip *sip, enum sip_transp tp)
+{
+	struct sa laddr;
+	int err;
+
+	if (sa_af(dst) != AF_INET6 || !sa_is_linklocal(dst))
+		return 0;
+
+	err = sip_transp_laddr(sip, &laddr, tp, dst);
+	if (err) {
+		DEBUG_WARNING("could not get laddr for %j\n", dst);
+		return err;
+	}
```

Comment:
will this work if you have two or more interfaces with IPv6-LL ?

example:
```
	inet6 fe80::40f7:f4ff:fe8f:20f0%awdl0 prefixlen 64 scopeid 0xa 
	inet6 fe80::f7e:ea28:f0b7:2bf9%utun0 prefixlen 64 scopeid 0xb 
```

`sip_transp_laddr` will return the first match.

if you are sending a SIP packet to an IPv6 device that is only reachable on network interface
number 2, how will this work ?

Reply:
In `transp_find()` we have:
```
		if (net_dst_source_addr_get(&dsttmp, &src))
			continue;

		if (!sa_cmp(&src, laddr, SA_ADDR))
			continue;
```
this will filter away the first IPv6ll address. It tests if there is a route from `laddr` to `dsttmp`. The second IPv6ll is returned.